routes/shopItemByStatus.js

```diff
@@ -1,25 +1,51 @@
 const express = require("express");
 const utility = require("../utility.js");
+const db = require("../server.js");
 
 findStatusRouter = express.Router();
 
 findStatusRouter.use(utility.readDatabaseMiddleware);
 
+//The original code
+// findStatusRouter.get("/", (req, res, next) => {
+//     const isCompleted = req.query.isCompleted || "";
+//     let groceryArray = req.database.shopItems;
+//     if (isCompleted === "true" || isCompleted === "false" || isCompleted === "") {
+//         if (isCompleted === "true") {
+//             groceryArray = utility.filterByStatus(groceryArray, true)
+//         } else if (isCompleted === "false") {
+//             groceryArray = utility.filterByStatus(groceryArray, false)
+//         }
+//         res.status(200);
+//         res.send(groceryArray);
+//     } else {
+//         res.status(400).send();
+//     }
+
+// });
+
+//My solution with sqlite
 findStatusRouter.get("/", (req, res, next) => {
-    const isCompleted = req.query.isCompleted || "";
-    let groceryArray = req.database.shopItems;
-    if (isCompleted === "true" || isCompleted === "false" || isCompleted === "") {
-        if (isCompleted === "true") {
-            groceryArray = utility.filterByStatus(groceryArray, true)
-        } else if (isCompleted === "false") {
-            groceryArray = utility.filterByStatus(groceryArray, false)
+    db.all("SELECT rowid, * FROM grocery_list", (err, row) => {
```

Comment:
better to add all fields needed for 'row' here instead of *. This way if the table changes later, this will still work as previously. In general * is rarely used in SELECT